### PR TITLE
feat(FN-1745): add 'Add to an existing payment' screen

### PR DESCRIPTION
**Pull request by sw-mattw**
## Introduction :pencil2:
Adds a new 'Add to an Existing Payment' screen to TFM UI which displays the record details table for the selected fee records.

## Resolution :heavy_check_mark:
- Added controller and routing for handling 'Add to an Existing Payment' POST requests from the 'Add Payment' screen
- New screen displays the fee record details table for the selected fee records
- Refactored shared functionality for mapping selected reported fees details into a separate helper function

## Miscellaneous :heavy_plus_sign:
Note that:
- Displaying available payments to be added in PR for FN-1727
- Adding the 'Continue' and 'Cancel' buttons and associated submission logic (in TFM UI, TFM API, and central API) to be covered in PR for FN-1749

New screen UI:
fn-1745_pr_add-to-an-existing-payment-screen

### CODE REVIEW

File: trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/selected-reported-fees-details-view-model-mapper.test.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { SelectedFeeRecordsDetailsResponseBody } from '../../../api-response-types';
+import { mapToSelectedReportedFeesDetailsViewModel } from '.';
+
+describe('selected-reported-fees-details-view-model-mapper', () => {
+  describe('mapToSelectedReportedFeesDetailsViewModel', () => {
+    it('should correctly map fee records to the selected reported fees details view model', () => {
```
Comment (RLavender98) [inline trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/selected-reported-fees-details-view-model-mapper.test.ts:6]: Can we drop the correctly from this test name please?

File: trade-finance-manager-ui/server/controllers/utilisation-reports/add-to-an-existing-payment/index.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  res.render('utilisation-reports/add-to-an-existing-payment.njk', context);
+
+export const addToAnExistingPayment = async (req: AddToAnExistingPaymentRequest, res: Response) => {
+  try {
+    const { user, userToken } = asUserSession(req.session);
+    const { reportId } = req.params;
+    const checkedCheckboxIds = getPremiumPaymentsCheckboxIdsFromObjectKeys(req.body);
+    const feeRecordIds = getFeeRecordIdsFromPremiumPaymentsCheckboxIds(checkedCheckboxIds);
+
+    // TODO FN-1749: Handle form submission.
```
Comment (RLavender98) [inline trade-finance-manager-ui/server/controllers/utilisation-reports/add-to-an-existing-payment/index.ts:28]: I think there is more of a chance of forgetting about this TODO then there is of forgetting to handle the form submission in the future ticket so I'd prefer it if we didn't merge this in.

File: trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/selected-reported-fees-details-view-model-mapper.test.ts (hunk 1 above)
Comment (RLavender98) [reply trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/selected-reported-fees-details-view-model-mapper.test.ts:6]: Actually, is this test adding anything that the controller tests don't? I'd actually be tempted to not add unit tests here if we aren't going to test it fully or test edge cases not covered in the controller tests. Otherwise if you do want to have unit tests for this then I would move the more detailed tests from the controllers e.g. the data-sort tests to this level and then just leave the high level tests in the controllers (e.g. `should fetch and map selected fee record details`).

**Review by RLavender98 — COMMENTED** [review]
Good work, just a few small comments 😄

File: trade-finance-manager-ui/server/types/view-models/add-to-an-existing-payment-view-model.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { SelectedReportedFeesDetailsViewModel } from './add-payment-view-model';
+import { BaseViewModel } from './base-view-model';
+
+export type AddToAnExistingPaymentViewModel = BaseViewModel & {
+  reportId: string;
+  bank: { name: string };
+  formattedReportPeriod: string;
+  reportedFeeDetails: SelectedReportedFeesDetailsViewModel;
```
[the remark this answers is not in the document]
Comment (sw-mattw) [reply trade-finance-manager-ui/server/types/view-models/add-to-an-existing-payment-view-model.ts:8]: Ah yes, pulled this out to its own file so it can be nicely shared

File: trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/selected-reported-fees-details-view-model-mapper.test.ts (hunk 1 above)
Comment (sw-mattw) [reply trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/selected-reported-fees-details-view-model-mapper.test.ts]: Removed this in favour of the unit tests on the controller

**Review by RLavender98 — APPROVED** [review]
